### PR TITLE
Dyn 2553 - Increase the code coverage for Logging folder

### PR DESCRIPTION
**Pull request by RobertGlobant20**
### Purpose

Increase the code coverage adding test cases for the DynamoCore/Logging folder 
https://jira.autodesk.com/browse/DYN-2553.
I've run the job in Jenkins5 and it finished correctly

### Declarations

Check these if you believe they are true

- [X] The codebase is in a better state after this PR
- [X] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [X] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@QilongTang 

### FYIs

@alfredo-pozo

### CODE REVIEW

**Comment by RobertGlobant20** [comment]
Here is the code coverage report generated by AxoCover just for the DynamoCore/Logging folder.

image

Thanks

File: test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                PathResolver = pathResolver,
+                StartInTestMode = false,
+                GeometryFactoryPath = preloader.GeometryFactoryPath,
+                Preferences = settings,
+                ProcessMode = TaskProcessMode.Synchronous
+            };
+        }
+
+        /// <summary>
+        /// This test method will validate that the AnalyticsService.OnWorkspaceAdded (CustomNodeWorkspaceModel) is executed
```
Comment (QilongTang) [inline test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs:36]: @RobertGlobant20 Is there other event tested? for example any node related ones?

File: test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                new FunctionNamePromptEventArgs
+                {
+                    Category = "Testing",
+                    Description = "",
+                    Name = "__AnalyticsServiceTest__",
+                    Success = true
+                }) as CustomNodeWorkspaceModel;
+
+            //Act
+            //This will execute the
```
Comment (QilongTang) [inline test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs:85]: incomplete comment

File: test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs (hunk 1 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs:36]: @QilongTang no, i think this is the only event in workspace (or nodes) tested as a part of my test cases added.

File: test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs (hunk 2 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs]: @QilongTang 
I modified the cs file and completed the comment and also added more description
see:
https://github.com/DynamoDS/Dynamo/pull/10546/commits/a79f52d69895469c578f472e757ebae264e1d32b

File: test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                new FunctionNamePromptEventArgs
+                {
+                    Category = "Testing",
+                    Description = "",
+                    Name = "__AnalyticsServiceTest__",
+                    Success = true
+                }) as CustomNodeWorkspaceModel;
+
+            //Act
+            //This will execute the
```
Comment (QilongTang) [inline test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs:67]: Same here incomplete comment

File: test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+            //First we try to get the value from the Windows Registry
+            var tryGetValue = Registry.GetValue(keyName, "InstrumentationGUID", null) as string;
+
+            if (tryGetValue != null)
+            {
+                //If the keyName already exists then we should remove it because the DynamoAnalyticsSession.GetUserID method will insert the value again
+                using (RegistryKey key = Registry.CurrentUser.OpenSubKey(subkey, true))
+                {
+                    key.DeleteValue("InstrumentationGUID");
```
Comment (QilongTang) [inline test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs:131]: Does this require some kind of permission elevation?

**Comment by QilongTang** [comment]
LGTM with one comment

File: test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs (hunk 4 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/Logging/DynamoAnalyticsClientTest.cs:131]: It's writing to the registry in the HKEY_CURRENT_USER key (current user logged), then we don't need extra permissions, also i ran the test cases in Visual Studio as a normal user and they ran successfully.

File: test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs (hunk 3 above)
Comment (RobertGlobant20) [reply test/DynamoCoreTests/Logging/AnalyticsServiceTest.cs]: @QilongTang 
Fixed, I completed the comment in the commit:
https://github.com/DynamoDS/Dynamo/pull/10546/commits/001c77502c11c8979b14f6d879577ba79a2eaf37

**Pull request merged**